### ISSUE TITLE
[Discussion] Declutter and simplify the repository

### ISSUE THREAD

**Issue by colinskow** [AngularClass/angular-starter#1735]
The simpler we keep things the lower the learning curve for new users and the less headache it is to add customizations.

It looks like there are a few "hacks" left over from the early days of Angular 2 that are no longer needed.

Examples:
- Most (or all) of `src/custom-typings.ts` is no longer needed.
- `src/app/environment.ts` seems a lot more complex than it needs to be.
- `package.json` scripts could be cleaned up and simplified a bit
- `+detail` and `+barrel` have nearly identical code. Do we need them both?
- Look for deprecated code that has been commented out for a while
- We should review for npm packages or Webpack plugins that are no longer needed or don't add value for the majority of users

Basically anything that can be removed without breaking tests or useful functionality should be.

cc: @gdi2290 @d3viant0ne

**Comment by SherlockStd** [comment]
I totally agree with you here ! :+1: 

Especially on:
- `src/app/environment.ts` seems a lot more complex than it needs to be.
- For `+detail` and `+barrel` we don't need both imao

**Comment by k1r0s** [comment]
I'm still wondering why do we need to define a `*.model.ts` in each +barrel | +detail child directory.

**Comment by gdi2290** [comment]
I'm going to simplify everything again

**Comment by AKlaus** [comment]
@gdi2290, should the issue be closed when the `custom-typings.ts` still contains all the references from the initial post?
